src/application/utils/transaction.ts

```diff
@@ -234,14 +266,26 @@ export function createTaxiTxFromTopup(
   const { selectedUtxos, changeOutputs } = coinSelector(throwErrorCoinSelector)(
     unspents,
     recipients.concat({
-      value: taxiTopup.assetAmount,
-      asset: taxiTopup.assetHash,
+      value: taxiReply.assetAmount,
+      asset: taxiReply.assetHash,
       address: '', // address is not useful for coinSelector
     }),
     changeAddressGetter
   );
-  const pset = addToTx(taxiTopup.partial, selectedUtxos, recipients.concat(changeOutputs));
-  return { pset, selectedUtxos };
+
+  const pset = Pset.fromBase64(taxiReply.topup.partial);
+  const initialNumberOfInputs = pset.inputs.length;
+  const updater = new Updater(pset);
+  updater.addInputs(selectedUtxos.map((u) => new CreatorInput(u.txid, u.vout)));
+  for (let i = 0; i < selectedUtxos.length; i++) {
+    updater.addInWitnessUtxo(i + initialNumberOfInputs, selectedUtxos[i].prevout);
+    updater.addInSighashType(i + initialNumberOfInputs, Transaction.SIGHASH_ALL);
+  }
+  updater.addOutputs(
+    recipients.concat(changeOutputs).map((r) => new CreatorOutput(r.asset, r.value, r.address, 0))
+  );
+
```

Comment:
you can add inputs and his witnessUtxos & sighash at the same time now with latest LiquidJS